src/safeds/data/tabular/containers/_table.py

```diff
@@ -1057,7 +1057,23 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
         2  5  6
         """
         if isinstance(rows, Table):
-            rows = rows.to_rows()
+            if rows.number_of_rows == 0:
+                return self
+            if self.number_of_columns == 0:
+                return rows
+            different_column_names = set(self.column_names) - set(rows.column_names)
+            if len(different_column_names) > 0:
+                raise UnknownColumnNameError(
+                    sorted(
+                        different_column_names,
+                        key={val: ix for ix, val in enumerate(self.column_names)}.__getitem__,
+                    ),
+                )
+
```

Comment:
```suggestion
            if self.number_of_rows == 0:
                return rows
            
```

Reply:
Well...
Seems like I was a bit too late :|

Reply:
@Marsmaennchen221 I can create a follow-up PR if you want

Reply:
Good point, though. Is this about line 1062 (`if self.number_of_columns == 0`?

Reply:
I'd check for either, as in either case it should be valid to do nothing